cmd/cli/plugin/package/ui.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	. "github.com/cppforlife/go-cli-ui/ui/table"
+	"github.com/spf13/cobra"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/component"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+type AdapterUI struct {
+	ui.WriterUI
+	outWriter    io.Writer
+	outputFormat string
+}
+
+func (adapterUI *AdapterUI) PrintLinef(pattern string, args ...interface{}) {
+	if strings.Contains(pattern, "Target cluster") {
```

Comment:
Could you please add a comment on what this function is supposed to do? Also the name of the function is so generic, yet it excludes some specific string like ```"Target cluster"```, and that is confusing

Reply:
Also the name of the function is so generic, yet it excludes some specific string like ```"Target cluster"```, and that is confusing

Reply:
>Could you please add a comment on what this function is meant to do?

Sure. Will do.

> Also the name of the function is so generic, yet it excludes some specific string like "Target cluster", and that is confusing

The idea is to override the kctrl ui using an adapter ui, so I can't use another name here, but the exclusion of certain strings simply based on a specific pattern bothers me as well. I will see how I can improve it.

Reply:
Ideally this should be done using the [tty](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/866424d03cea80e5cb35598f2ab449defe007d87/cli/pkg/kctrl/cmd/ui_flags.go#L23) flag, but there is an issue with the tty flag where in it is always set to true unless the output is being piped. The issue exists in all the carvel tools and we will be fixing it soon.

Reply:
Once the tty issue is fixed, we should be able to remove the `PrintLinef` "override".